src/controller/python/chip/ChipCluster.py

```diff
@@ -0,0 +1,1619 @@
+                    ("str", "sceneName"),
+                    ("int", "clusterId"),
+                    ("int", "length"),
+                    ("int", "value"),
+                ],
+                "GetSceneMembership": [
+                    ("int", "groupId"),
+                ],
+                "RecallScene": [
+                    ("int", "groupId"),
+                    ("int", "sceneId"),
+                    ("int", "transitionTime"),
+                ],
+                "RemoveAllScenes": [
+                    ("int", "groupId"),
+                ],
+                "RemoveScene": [
+                    ("int", "groupId"),
+                    ("int", "sceneId"),
+                ],
+                "StoreScene": [
+                    ("int", "groupId"),
+                    ("int", "sceneId"),
+                ],
+                "ViewScene": [
+                    ("int", "groupId"),
+                    ("int", "sceneId"),
+                ],
+            },
+            "TemperatureMeasurement": {
+            },
+        }
+
+    def PrepareCommand(self, cluster: str, command: str, endpoint: int, groupid: int, args):
+        func = getattr(self, "Cluster{}_Command{}".format(cluster, command))
+        if not func:
+            raise UnknownCluster(cluster)
+        func(endpoint, groupid, args)
+
+    def ClusterBarrierControl_CommandBarrierControlGoToPercent(self, ZCLendpoint: int, ZCLgroupid: int, args):
+        totArgs = 0 + 1
+        if len(args) != totArgs:
+            raise InvalidArgumentCount(totArgs, len(args))
+        argCnt = 0
+        percentOpen = int(args[argCnt])
+        argCnt += 1
+        self._ChipStack.Call(
+            lambda: self._dmLib.chip_ime_AppendCommand_BarrierControl_BarrierControlGoToPercent(
+                ZCLendpoint, ZCLgroupid, percentOpen
+            )
+        )
+
+    def ClusterBarrierControl_CommandBarrierControlStop(self, ZCLendpoint: int, ZCLgroupid: int, args):
+        totArgs = 0
+        if len(args) != totArgs:
+            raise InvalidArgumentCount(totArgs, len(args))
+        argCnt = 0
+        self._ChipStack.Call(
+            lambda: self._dmLib.chip_ime_AppendCommand_BarrierControl_BarrierControlStop(
+                ZCLendpoint, ZCLgroupid
+            )
+        )
+
+    def ClusterBasic_CommandResetToFactoryDefaults(self, ZCLendpoint: int, ZCLgroupid: int, args):
+        totArgs = 0
+        if len(args) != totArgs:
+            raise InvalidArgumentCount(totArgs, len(args))
+        argCnt = 0
+        self._ChipStack.Call(
+            lambda: self._dmLib.chip_ime_AppendCommand_Basic_ResetToFactoryDefaults(
+                ZCLendpoint, ZCLgroupid
+            )
+        )
+
+    def ClusterColorControl_CommandMoveColor(self, ZCLendpoint: int, ZCLgroupid: int, args):
+        totArgs = 0 + 1 + 1 + 1 + 1
+        if len(args) != totArgs:
+            raise InvalidArgumentCount(totArgs, len(args))
+        argCnt = 0
```

Comment:
This indexing feels a bit odd in python. I wonder if this would be more pythonic (because shorter):
```py
argsIterator = iter(args)

rateX = int(next(argsIterator))
rateY = int(next(argsIterator))
optionsMask = int(next(argsIterator))
optionsOverride = int(next(argsIterator))
```

as a human I would do:

```py
rateX, rateY, optionsMask, optionsOverride = map(int, args)
```

however I imagine that is not scalable of different type of arguments.

Reply:
Modified api, now arguments are passed as function arguments directly.